jetty-servlet/src/test/java/org/eclipse/jetty/servlet/AsyncServletIOTest.java

```diff
@@ -801,6 +804,106 @@ public void onAllDataRead() throws IOException
         }
     }
     
+    
+
+    @Test
+    public void testTimeoutWhilePendingWrite() throws Exception
```

Comment:
Should be moved to the `tests/test-http-client-transport` module to verify both HTTP and HTTP/2 behavior.
There, there already exist a `AsyncIOServletTest` that you can use.